crates/client/eth/src/state_update.rs

```diff
@@ -0,0 +1,261 @@
+        node_bindings::Anvil,
+        providers::{ProviderBuilder, RootProvider},
+        sol,
+        transports::http::{Client, Http},
+    };
+    use dc_db::{block_db::ChainInfo, DatabaseService};
+    use dc_metrics::{block_metrics::block_metrics::BlockMetrics, MetricsService};
+    use starknet_types_core::felt::Felt;
+    use tokio::time::timeout;
+    use url::Url;
+
+    sol!(
+        #[allow(missing_docs)]
+        #[sol(rpc, bytecode="6080604052348015600e575f80fd5b506101618061001c5f395ff3fe608060405234801561000f575f80fd5b5060043610610029575f3560e01c80634185df151461002d575b5f80fd5b610035610037565b005b5f7f0639349b21e886487cd6b341de2050db8ab202d9c6b0e7a2666d598e5fcf81a690505f620a1caf90505f7f0279b69383ea92624c1ae4378ac7fae6428f47bbd21047ea0290c3653064188590507fd342ddf7a308dec111745b00315c14b7efb2bdae570a6856e088ed0c65a3576c8383836040516100b9939291906100f6565b60405180910390a1505050565b5f819050919050565b6100d8816100c6565b82525050565b5f819050919050565b6100f0816100de565b82525050565b5f6060820190506101095f8301866100cf565b61011660208301856100e7565b61012360408301846100cf565b94935050505056fea2646970667358221220fbc6fd165c86ed9af0c5fcab2830d4a72894fd6a98e9c16dbf9101c4c22e2f7d64736f6c634300081a0033")]
+        contract DummyContract {
+            event LogStateUpdate(uint256 globalRoot, int256 blockNumber, uint256 blockHash);
+
+            function fireEvent() public {
+                uint256 globalRoot = 2814950447364693428789615812443623689251959344851195711990387747563915674022;
+                int256 blockNumber = 662703;
+                uint256 blockHash = 1119674286844400689540394420005977072742999649767515920196535047615668295813;
+
+                emit LogStateUpdate(globalRoot, blockNumber, blockHash);
+            }
+        }
+    );
+
+    const ETH_URL: &str = "https://eth.merkle.io";
+    const L1_BLOCK_NUMBER: u64 = 20395662;
+    const L2_BLOCK_NUMBER: u64 = 662703;
+    const TEST_TIMEOUT: u64 = 5; // Timeout in seconds
+    const EVENT_PROCESSING_TIME: u64 = 2; // Time to allow for event processing in seconds
+    const ADDITIONAL_WAIT_TIME: u64 = 1; // Additional wait time in seconds
+
+    /// Test the event subscription and state update functionality
+    ///
+    /// This test performs the following steps:
+    /// 1. Sets up a mock Ethereum environment using Anvil
+    /// 2. Initializes necessary services (Database, Metrics)
+    /// 3. Deploys a dummy contract and sets up an Ethereum client
+    /// 4. Starts listening for state updates
+    /// 5. Fires an event from the dummy contract
+    /// 6. Waits for event processing and verifies the block number
+    #[tokio::test]
+    async fn test_event_subscription() {
+        // Set up chain info
+        let chain_info =
+            ChainInfo { chain_id: Felt::from_dec_str("11153111").unwrap(), chain_name: "testing".to_string() };
+
+        // Set up database paths
+        let current_dir = env::current_dir().expect("Failed to get current directory");
```

Comment:
can you check the temp dir thing if it can be used here?

Reply:
added